File: v131016/TeeNew/Margins_Units.pas
unit Margins_Units;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QComCtrls, QStdCtrls, QExtCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ComCtrls, StdCtrls, ExtCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, TeCanvas;

type
  TChartMarginUnits = class(TBaseForm)
    Label1: TLabel;
    ComboBox1: TComboBox;
    Label2: TLabel;
    Edit1: TEdit;
    UpDown1: TUpDown;
    Label3: TLabel;
    procedure ComboBox1Change(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Edit1Change(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.dfm}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TChartMarginUnits.ComboBox1Change(Sender: TObject);
begin
  if ComboBox1.ItemIndex=0 then
  begin
    Chart1.MarginUnits:=muPercent;
    Label3.Caption:='%';
  end
  else
  begin
    Chart1.MarginUnits:=muPixels;
    Label3.Caption:='pixels';
  end;
end;

procedure TChartMarginUnits.FormCreate(Sender: TObject);
begin
  inherited;
  ComboBox1.ItemIndex:=1;
  ComboBox1Change(Self);
end;

procedure TChartMarginUnits.Edit1Change(Sender: TObject);
begin
  Chart1.MarginLeft  :=UpDown1.Position;
  Chart1.MarginTop   :=UpDown1.Position;
  Chart1.MarginRight :=UpDown1.Position;
  Chart1.MarginBottom:=UpDown1.Position;
end;

initialization
  RegisterClass(TChartMarginUnits);
end.

File: TeeNew/Builder2006/Margins_Units.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Margins_Units.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TChartMarginUnits *ChartMarginUnits;
//---------------------------------------------------------------------------
__fastcall TChartMarginUnits::TChartMarginUnits(TComponent* Owner)
  : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TChartMarginUnits::ComboBox1Change(TObject *Sender)
{
  if (ComboBox1->ItemIndex == 0)
  {
    Chart1->MarginUnits = muPercent;
    Label3->Caption = "%";
  }
  else
  {
    Chart1->MarginUnits = muPixels;
    Label3->Caption = "pixels";
  }
}
//---------------------------------------------------------------------------

void __fastcall TChartMarginUnits::FormCreate(TObject *Sender)
{
  ComboBox1->ItemIndex = 1;
  ComboBox1Change(this);
}
//---------------------------------------------------------------------------

void __fastcall TChartMarginUnits::Edit1Change(TObject *Sender)
{
  Chart1->MarginLeft = UpDown1->Position;
  Chart1->MarginTop = UpDown1->Position;
  Chart1->MarginRight = UpDown1->Position;
  Chart1->MarginBottom = UpDown1->Position;
}
//---------------------------------------------------------------------------

